### DISCUSSION TITLE
PowerWall New Features

### DISCUSSION THREAD

**Discussion by dswedge** [tdorssers/TeslaPy#79] [Ideas]
As of Tesla App v 4.9.0, perhaps before, settings for two new PowerWall capabilities were added for some owners:

Grid Charging (No/Yes)

Energy Exports (Solar/Everything)

Someone posted "endpoints" on Tesla Motors Club:

https://owner-api.teslamotors.com/api/1/energy_sites/{}/grid_import_export
Payload options:
"customer_preferred_export_rule": "battery_ok"
"customer_preferred_export_rule": "pv_only"
"disallow_charge_from_grid_with_solar_installed": false
"disallow_charge_from_grid_with_solar_installed": true

Some users also have a No Export option.  I don't have that one, so I am not certain how it appears.

Are these yet included in TeslaPy?

Thank you!
SW

**Comment by tdorssers** [comment]
That endpoint is called `ENERGY_SITE_IMPORT_EXPORT_CONFIG` and is included in TeslaPy v2.5.0. Feel free to implement additional Powerwall methods and send a pull request.

**Comment by purcell-lab** [comment]
Just playing with the new FleetAPI and it looks like there is an additional payload for customer_preferred_export_rule, which is never.

https://developer.tesla.com/docs/fleet-api#grid_import_export
